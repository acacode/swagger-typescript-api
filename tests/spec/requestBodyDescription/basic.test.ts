import * as fs from "node:fs/promises";
import * as os from "node:os";
import * as path from "node:path";

import { afterAll, beforeAll, describe, expect, test } from "vitest";

import { generateApi } from "../../../src/index.js";

describe("basic", async () => {
  let tmpdir = "";

  beforeAll(async () => {
    tmpdir = await fs.mkdtemp(path.join(os.tmpdir(), "swagger-typescript-api"));
  });

  afterAll(async () => {
    await fs.rm(tmpdir, { recursive: true });
  });

  test("requestBodyDescription", async () => {
    await generateApi({
      fileName: "schema",
      input: path.resolve(import.meta.dirname, "schema.yml"),
      output: tmpdir,
      silent: true,
      generateClient: false,
      generateRouteTypes: false,
      extractResponseBody: true,
      extractRequestParams: true,
      extractRequestBody: true,
      extractResponseError: true,
      extractResponses: true,
      extractEnums: true,
    });

    const content = await fs.readFile(path.join(tmpdir, "schema.ts"), {
      encoding: "utf8",
    });

    expect(content).toMatchSnapshot();
  });
});
