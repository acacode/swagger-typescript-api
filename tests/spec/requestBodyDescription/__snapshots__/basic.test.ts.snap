// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`basic > requestBodyDescription 1`] = `
"/* eslint-disable */
/* tslint:disable */
// @ts-nocheck
/*
 * ---------------------------------------------------------------
 * ## THIS FILE WAS GENERATED VIA SWAGGER-TYPESCRIPT-API        ##
 * ##                                                           ##
 * ## AUTHOR: acacode                                           ##
 * ## SOURCE: https://github.com/acacode/swagger-typescript-api ##
 * ---------------------------------------------------------------
 */

/** FODSFDSF DSF DSF */
export type CreateKeksPayload = object;

export type CreateKeksData = object;
"
`;
